src/Avalonia.Controls/Notifications/WindowNotificationManager.cs

```diff
@@ -0,0 +1,170 @@
+
+        public int MaxItems
+        {
+            get { return GetValue(MaxItemsProperty); }
+            set { SetValue(MaxItemsProperty, value); }
+        }
+
+        public WindowNotificationManager(Window host)
+        {
+            if (VisualChildren.Count != 0)
+            {
+                Install(host);
+            }
+            else
+            {
+                IDisposable stopListening = null;
+
+                stopListening = Observable.FromEventPattern<TemplateAppliedEventArgs>(host, nameof(host.TemplateApplied))
+                    .Subscribe(_ =>
+                    {
+                        stopListening.Dispose();
+
+                        Install(host);
+                    });
+            }
+        }
+
+        static WindowNotificationManager()
+        {
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.TopLeft, ":topleft");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.TopRight, ":topright");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomLeft, ":bottomleft");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomRight, ":bottomright");
+
+            HorizontalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.HorizontalAlignment.Stretch);
+            VerticalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.VerticalAlignment.Stretch);
+        }
+
+        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        {
+            base.OnTemplateApplied(e);
+
+            var itemsControl = e.NameScope.Find<Panel>("PART_Items");
+            _items = itemsControl?.Children;
+        }
+
+        public void Show(INotification content)
+        {
+            Show(content as object);
+        }
+
+        public async void Show(object content)
+        {
+            var notification = content as INotification;
+
+            var notificationControl = new NotificationCard
+            {
+                Content = content
+            };
+
+            if (notification != null)
+            {
+                notificationControl.NotificationClosed += (sender, args) => notification.OnClose?.Invoke();
+            }
+
+            notificationControl.NotificationClosed += OnNotificationClosed;
+
+            notificationControl.PointerPressed += (sender, args) =>
+            {
+                if (notification != null && notification.OnClick != null)
+                {
+                    notification.OnClick.Invoke();
+                    (sender as NotificationCard)?.Close();
+                }
+            };
+
+            lock (_items)
```

Comment:
Is this lock needed? Afaics `_items` will only ever be:

- Added to in `Show(object)` which is guaranteed to be on the main thread because it creates a `NotificationCard`
- Removed from in `OnNotificationClosed` which is guaranteed to be on the main thread because it's an Avalonia event handler

Reply:
True I think I was thinking of allowing people to call Show from any thread, will remove.